File: src/common/impl/gn_build_example.cc
#include "src/common/include/gn_build_example.h"
#include "src/common/module/base64/Base64.h"
#include <string>

const char * gn_build_example_get_version() {
#ifndef GBE_VERSION
#define GBE_VERSION "Unknown"
#endif
    return GBE_VERSION;
}

const char * gn_build_example_get_message(enum gn_build_example_message_type type) {

#ifndef GBE_CUSTOM_MESSAGE
#define GBE_CUSTOM_MESSAGE "Hello World!" // Default message
#endif
    const char *raw_message = GBE_CUSTOM_MESSAGE;
    if (type == gn_build_example_message_type_raw) {
        return raw_message;
    } else if (type == gn_build_example_message_type_base_64) {
        int raw_length = strlen(raw_message);
        int encoded_length = base64_enc_len(raw_length);
        char *encoded_message = new char[encoded_length];
        base64_encode(encoded_message, (char *)raw_message, raw_length);
        return encoded_message;
    } else {
        return "Unknown type";
    }
}

File: src/platform/darwin/GNBuildExample/impl/GNBuildExample.m
//
//  GNBuildExample.m
//  GNBuildExample
//
//  Created by Patrick Fu on 2021/2/28.
//  Copyright © 2021 Patrick Fu. All rights reserved.
//

#import "src/platform/darwin/GNBuildExample/include/GNBuildExample.h"
#include "src/common/include/gn_build_example.h"

@interface GNBuildExample ()

@end

@implementation GNBuildExample

+ (NSString *)getVersion {
    return [NSString stringWithUTF8String:gn_build_example_get_version()];
}

+ (NSString *)getMessage:(GNBuildExampleMessageType)type {
    const char * message = gn_build_example_get_message((enum gn_build_example_message_type)type);
    return [NSString stringWithUTF8String:message];
}

@end
